src/main/java/io/papermc/lib/features/offlineplayers/GetOfflinePlayerNoCacheOption.java

```diff
@@ -0,0 +1,15 @@
+package io.papermc.lib.features.offlineplayers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+
+public class GetOfflinePlayerNoCacheOption implements GetOfflinePlayer {
+    @Override
+    @Nullable
+    public OfflinePlayer getOfflinePlayer(@Nonnull String name, boolean makeWebRequest) {
+        return Bukkit.getOfflinePlayer(name);
```

Comment:
reflection can be used here to check if the profile cache returns null for the given username (`UserCache#c#get` mapping has been consistent since 1.8.8). That way the `makeWebRequest` parameter can be respected even if we're on spigot.

(core team may have input on this)

Reply:
Hmm, that would certainly be a neat idea.
I will look into it but also wait and see what others think about this.

Reply:
So I had a look into it and `UserCache#c` seems to refer to a static boolean on Spigot 1.16.4 (tested using `git-Spigot-05da6fa-b0c6dfe`).
So unless, that was a typo the mapping does not seem to be as consistent as expected.
The actual `Map<String, UserCacheEntry>` is found under `UserCache#d` on the build I investigated.

There is a `UserCache#getProfile(String)` method though, which returns a nullable GameProfile for a given username from the cache (unless expired).
I would need to look up how long this method existed though.